src/js_types.hpp

```diff
@@ -351,11 +351,12 @@ REALM_JS_INLINE void set_internal(const typename T::Object &object, typename Cla
 
 template<typename T>
 inline bool Value<T>::is_valid_for_property(ContextType context, const ValueType &value, const Property& prop)
-{
-    if (prop.is_nullable && (is_null(context, value) || is_undefined(context, value))) {
+{ 
+    if (is_nullable(prop.type) && (is_null(context, value) || is_undefined(context, value))) {
         return true;
     }
 
+    //lubo
     using PropertyType = realm::PropertyType;
     switch (prop.type) {
```

Comment:
`PropertyType::Array` and `PropertyType::LinkingObjects` should be removed from this switch, and instead tested for with `is_array(prop.type)` then testing for `PropertyType::Object` and `PropertyType::LinkingObjects` respectively.